NuRadioReco/modules/efieldGalacticNoiseAdder.py

```diff
@@ -0,0 +1,153 @@
+                return
+            last_freqs = field.get_frequencies()
+
+        freqs = last_freqs
+        d_f = freqs[2] - freqs[1]
+
+        # If we cache the antenna pattern, we need to make sure that the frequencies have not changed
+        # between stations. If they have, we need to clear the cache.
+        if self.__caching:
+            if self.__freqs is None:
+                self.__freqs = freqs
+            else:
+                if not np.allclose(self.__freqs, freqs, rtol=0, atol=0.01 * units.MHz):
+                    self.__freqs = freqs
+                    self._get_cached_antenna_response.cache_clear()
+                    logger.warning(
+                        "Frequencies have changed. Clearing antenna response cache. "
+                        "(If this happens often, something might be wrong...")
+
+        if passband is None:
+            passband = [10 * units.MHz, 1000 * units.MHz]
+
+        passband_filter = (freqs > passband[0]) & (freqs < passband[1])
+
+        site_latitude, site_longitude = detector.get_site_coordinates(station.get_id())
+        station_time = station.get_station_time()
+
+        local_coordinates = get_local_coordinates((site_latitude, site_longitude), station_time, self.__n_side)
+
+        # n_ice = ice.get_refractive_index(-0.01, detector.get_site(station.get_id()))
+        n_air = ice.get_refractive_index(depth=1, site=detector.get_site(station.get_id()))
+        c_vac = scipy.constants.c * units.m / units.s
+
+        field_spectra = {}
+        for field in station.get_electric_fields():
+            field_spectra[field.get_unique_identifier()] = field.get_frequency_spectrum()
+
+        for i_pixel in range(healpy.pixelfunc.nside2npix(self.__n_side)):
+            azimuth = local_coordinates[i_pixel].az.rad
+            zenith = np.pi / 2. - local_coordinates[i_pixel].alt.rad
+
+            if zenith > 90. * units.deg:
+                continue
+
+            temperature_interpolator = scipy.interpolate.interp1d(
+                self.__interpolation_frequencies, np.log10(self.__noise_temperatures[:, i_pixel]), kind='quadratic')
+            noise_temperature = np.power(10, temperature_interpolator(freqs[passband_filter]))
+
+            # calculate spectral radiance of radio signal using rayleigh-jeans law
+            spectral_radiance = (2. * (scipy.constants.Boltzmann * units.joule / units.kelvin)
+                                 * freqs[passband_filter] ** 2 * noise_temperature * self.solid_angle / c_vac ** 2)
+            spectral_radiance[np.isnan(spectral_radiance)] = 0
+
+            # calculate radiance per energy bin
+            spectral_radiance_per_bin = spectral_radiance * d_f
+
+            # calculate electric field per energy bin from the radiance per bin
+            efield_amplitude = np.sqrt(
+                spectral_radiance_per_bin / (c_vac * scipy.constants.epsilon_0 * (
+                        units.coulomb / units.V / units.m))) / d_f
```

Comment:
Shall we add this code block in the utilities? Would be also useful for modules which Ruben is working on.